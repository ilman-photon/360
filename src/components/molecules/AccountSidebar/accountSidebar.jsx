import { Collapse, Fade, Stack, Typography } from "@mui/material";
import { useRouter } from "next/router";
import { useState } from "react";
import SidebarLink from "../../atoms/SidebarLink/sidebarLink";
import styles from "./accountSidebar.module.scss";
import AccountCircleOutlinedIcon from "@mui/icons-material/AccountCircleOutlined";
import DescriptionOutlinedIcon from "@mui/icons-material/DescriptionOutlined";
import AssignmentTurnedInOutlinedIcon from "@mui/icons-material/AssignmentTurnedInOutlined";
import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
import KeyboardArrowRightOutlinedIcon from "@mui/icons-material/KeyboardArrowRightOutlined";
import KeyboardArrowDownOutlinedIcon from "@mui/icons-material/KeyboardArrowDownOutlined";
import { colors } from "../../../styles/theme";
import { useEffect } from "react";

export const AccountSidebar = ({ ...props }, ref) => {
  const [activeSidebarChild, setActiveSidebarChild] = useState(null);

  const router = useRouter();
  const sidebarLinks = [
    {
      label: "Profile Information",
      href: "/patient/account/profile-info",
      icon: <AccountCircleOutlinedIcon />,
    },
    {
      label: "Care Plan",
      href: "#",
      icon: <AccountCircleOutlinedIcon />,
    },
    { label: "Prescriptions", href: "#", icon: <AccountCircleOutlinedIcon /> },
    {
      label: "Test & Lab Results",
      href: "#",
      icon: <AssignmentTurnedInOutlinedIcon />,
    },
    {
      label: "Documents",
      child: [
        {
          label: "Intake Forms",
          href: "/patient/account/documents/intake-forms",
        },
        {
          label: "Insurance documents",
          href: "/patient/account/insurance-info",
        },
      ],
      icon: <DescriptionOutlinedIcon />,
    },
    // { label: "Insurance documents", href: "/patient/account/insurance-info" },
    {
      label: "Multi factor authentication",
      href: "#",
      icon: <LockOutlinedIcon />,
    },
  ];

  const toggleSidebarChild = (payload) => {
    if (activeSidebarChild !== payload) {
      setActiveSidebarChild(payload);
    } else {
      setActiveSidebarChild(null);
    }
  };

  useEffect(() => {
    setActiveSidebarChild(router.pathname.split("/")[3]);
  }, [router]);

  return (
    <div className={styles.sidebarWrapper}>
      <Stack className={styles.sidebarLinkWrapper} spacing={2}>
        {sidebarLinks.map((link, idx) => {
          return link.href ? (
            <Stack
              flexDirection="row"
              alignItems="center"
              color={colors.iconGrey}
            >
              {link.icon}
              <SidebarLink
                router={router}
                key={idx}
                href={link.href}
                child={link.child}
              >
                {link.label}
              </SidebarLink>
            </Stack>
          ) : (
            <Stack>
              <Stack
                flexDirection="row"
                alignItems="center"
                color={colors.iconGrey}
                sx={{ cursor: "pointer" }}
                onClick={() => toggleSidebarChild("documents")}
              >
                {link.icon}
                <Typography
                  variant="allVariants"
                  className={styles.sidebarLink}
                >
                  {link.label}
                </Typography>
                {activeSidebarChild === "documents" ? (
                  <KeyboardArrowDownOutlinedIcon
                    sx={{ width: 23, height: 23, ml: "auto" }}
                  />
                ) : (
                  <KeyboardArrowRightOutlinedIcon
                    sx={{ width: 23, height: 23, ml: "auto" }}
                  />
                )}
              </Stack>

              <Collapse
                in={activeSidebarChild === "documents"}
                unmountOnExit
                sx={{ ml: 2 }}
              >
                <Stack sx={{ ml: 2 }}>
                  {link.child.map((item, childIdx) => {
                    return (
                      <SidebarLink
                        router={router}
                        key={childIdx}
                        href={item.href}
                        child={item.child}
                      >
                        {item.label}
                      </SidebarLink>
                    );
                  })}
                </Stack>
              </Collapse>
            </Stack>
          );
        })}
      </Stack>
    </div>
  );
};

export default AccountSidebar;
