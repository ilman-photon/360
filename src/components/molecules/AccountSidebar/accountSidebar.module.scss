.sidebarWrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  border-radius: 4px;
  height: 100%;
  font-family: "Museo Sans", sans-serif;
  background-color: #f4f4f4;
}

.sidebarLinkWrapper {
  padding: 40px 16px 40px 24px;
  // margin-left: 240px;

  @media (max-width: 1536px) {
    width: 100%;
  }
  @media (min-width: 1535px) {
    // margin-left: auto;
    width: fit-content;
  }
}

.sidebarLink {
  display: flex;
  align-items: center;
  padding: 13px 16px;
  gap: 10px;

  // width: 324px;
  // height: 46px;
}
