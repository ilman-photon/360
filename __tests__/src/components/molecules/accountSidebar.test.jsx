import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import AccountSidebar from "../../../../src/components/molecules/AccountSidebar/accountSidebar";

window.scrollTo = jest.fn();

describe("AccountSidebar Components", () => {
  let container;
  beforeEach(() => {
    container = render(<AccountSidebar />);
  });

  it("AccountSidebar render", () => {
    expect(container.getByText(/Profile Information/i)).toBeInTheDocument();
    // expect(container.getByText("Care Plan")).toBeInTheDocument();
    // expect(container.getByText("Prescriptions")).toBeInTheDocument();
    // expect(container.getByText("Test & Lab Results")).toBeInTheDocument();
    expect(container.getByText("Insurance documents")).toBeInTheDocument();
    expect(
      container.getByText("Multi factor authentication")
    ).toBeInTheDocument();

    setTimeout(() => {
      expect("Profile Information").toBeVisible();
      // expect("Care Plan").toBeVisible();
      // expect("Prescriptions").toBeVisible();
      // expect("Test & Lab Results").toBeVisible();
      expect("Insurance documents").toBeVisible();
      expect("Multi factor authentication").toBeVisible();
    }, 500);
  });
});
