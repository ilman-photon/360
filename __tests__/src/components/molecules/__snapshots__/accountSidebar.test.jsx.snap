// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSidebar Components AccountSidebar render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebarWrapper css-1v482q1-MuiStack-root"
      >
        <a
          class="sidebarLink"
          href="/patient/account/profile-info"
          style="border-color: #003b4a;"
        >
          Profile Information
        </a>
        <a
          class="sidebarLink"
          href="#"
          style="border-color: #003b4a;"
        >
          Financial Information
        </a>
        <a
          class="sidebarLink"
          href="#"
          style="border-color: #003b4a;"
        >
          Toggle accounts
        </a>
        <a
          class="sidebarLink"
          href="#"
          style="border-color: #003b4a;"
        >
          Merge accounts
        </a>
        <a
          class="sidebarLink"
          href="#"
          style="border-color: #003b4a;"
        >
          Prescriptions
        </a>
        <a
          class="sidebarLink"
          href="/patient/account/insurance-info"
          style="border-color: #003b4a;"
        >
          Insurance documents
        </a>
        <a
          class="sidebarLink"
          href="#"
          style="border-color: #003b4a;"
        >
          Multi factor authentication
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebarWrapper css-1v482q1-MuiStack-root"
    >
      <a
        class="sidebarLink"
        href="/patient/account/profile-info"
        style="border-color: #003b4a;"
      >
        Profile Information
      </a>
      <a
        class="sidebarLink"
        href="#"
        style="border-color: #003b4a;"
      >
        Financial Information
      </a>
      <a
        class="sidebarLink"
        href="#"
        style="border-color: #003b4a;"
      >
        Toggle accounts
      </a>
      <a
        class="sidebarLink"
        href="#"
        style="border-color: #003b4a;"
      >
        Merge accounts
      </a>
      <a
        class="sidebarLink"
        href="#"
        style="border-color: #003b4a;"
      >
        Prescriptions
      </a>
      <a
        class="sidebarLink"
        href="/patient/account/insurance-info"
        style="border-color: #003b4a;"
      >
        Insurance documents
      </a>
      <a
        class="sidebarLink"
        href="#"
        style="border-color: #003b4a;"
      >
        Multi factor authentication
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
